### ISSUE TITLE
short[] causes broken code

### ISSUE THREAD

**Issue by rekire** [mcharmas/android-parcelable-intellij-plugin#59]
I already read in the readme.md that you don't support that type, but the generated code does not compile. If you have some time please fix that.

**Comment by DariusL** [comment]
That's how "not supported" is implemented. You can generate the entire class and change the parts that read and write your array. What other solution would you recommend?

**Comment by rekire** [comment]
I would prefer if the generated code would just work^^

Here is the generated code from my own library:

```
// read from the parcable
int[] shortArrayAsIntArray = in.createIntArray();
shortArray = new short[shortArrayAsIntArray.length];
for(int i = 0; i < shortArrayAsIntArray.length; i++) {
    shortArray[i] = (short)shortArrayAsIntArray[i];
}

// write to the parcable
int[] shortArrayAsIntArray = new int[shortArray.length];
for(int i = 0; i < shortArrayAsIntArray.length; i++) {
    shortArrayAsIntArray[i] = (int)shortArray[i];
}
dest.writeIntArray(shortArrayAsIntArray);
```

Currently will be this code generated which won't compile:

```
// write to the parcable
dest.writeParcelable(this.shortArray, flags);
// read from the parcable
this.shortArray = in.readParcelable(short[].class.getClassLoader());
```

**Comment by DariusL** [comment]
I agree that the feature is useful, I'm just saying that the best failure is an early failure. Your code would work, I don't know if this plugin can generate a utils class with methods for conversion, though. So you'll have to keep this code in every task.

Other than that, you could probably pack two shorts into an int for memory efficiency, but you'll need benchmarks to check if that helps.

**Comment by rekire** [comment]
If you talk about memory efficiency keep in mind that every boolean is stored as byte or even as integer. They can been packed much more efficiency.
